### PR TITLE
Bridges - changes for Bridges V2 - compact proofs

### PR DESCRIPTION
**Pull request by bkontur**
TBD:

### CODE REVIEW

File: bridges/primitives/messages/src/target_chain.rs [hunk 1]
```diff
@@ -16,17 +16,47 @@
...
+///
+/// - lane id;
+///
+/// - nonces (inclusive range) of messages which are included in this proof.
+#[derive(Clone, Decode, Encode, Eq, PartialEq, RuntimeDebug, TypeInfo)]
+pub struct FromBridgedChainMessagesProof<BridgedHeaderHash> {
+	/// Hash of the finalized bridged header the proof is for.
+	pub bridged_header_hash: BridgedHeaderHash,
+	/// The proved storage containing the messages being delivered.
+	pub storage: UnverifiedStorageProof,
```
Comment (bkontur) [inline bridges/primitives/messages/src/target_chain.rs:45]: @serban300 could you please check/correct this doc? It says "proved" but `UnverifiedStorageProof` is used here, is that ok?

File: bridges/primitives/messages/src/target_chain.rs [hunk 2]
```diff
@@ -16,17 +16,47 @@
...
 use sp_std::{collections::btree_map::BTreeMap, fmt::Debug, marker::PhantomData, prelude::*};
 
+/// Messages proof from bridged chain.
+///
+/// It contains everything required to prove that bridged (source) chain has
+/// sent us some messages:
+///
+/// - hash of finalized header;
+///
+/// - storage proof of messages and (optionally) outbound lane state;
```
Comment (bkontur) [inline bridges/primitives/messages/src/target_chain.rs:35]: @serban300 what is "(optionally) outbound lane state" here?

Comment (serban300) [reply bridges/primitives/messages/src/target_chain.rs:35]: I think it's the `outbound_lane_data`. I'm not completely sure.

File: bridges/primitives/messages/src/target_chain.rs (hunk 1 above)
Comment (serban300) [reply bridges/primitives/messages/src/target_chain.rs:45]: Yes, I agree it's confusing. The `UnverifiedStorageProof` is the storage + the prove for it from what I remember. So it's "proved". But we can rephrase it to something like "The storage containing the messages being delivered."

**Comment by bkontur** [comment]
bot bench cumulus-bridge-hubs --runtime=bridge-hub-rococo --pallet=pallet_bridge_grandpa
bot bench cumulus-bridge-hubs --runtime=bridge-hub-rococo --pallet=pallet_bridge_parachains
bot bench cumulus-bridge-hubs --runtime=bridge-hub-rococo --pallet=pallet_bridge_messages
bot bench cumulus-bridge-hubs --runtime=bridge-hub-rococo --pallet=pallet_bridge_relayers
bot bench cumulus-bridge-hubs --runtime=bridge-hub-westend --pallet=pallet_bridge_grandpa
bot bench cumulus-bridge-hubs --runtime=bridge-hub-westend --pallet=pallet_bridge_parachains
bot bench cumulus-bridge-hubs --runtime=bridge-hub-westend --pallet=pallet_bridge_messages
bot bench cumulus-bridge-hubs --runtime=bridge-hub-westend --pallet=pallet_bridge_relayers

**Comment by bkontur** [comment]
bot clean

**Comment by bkontur** [comment]
bot clean

**Comment by bkontur** [comment]
bot bench cumulus-bridge-hubs --runtime=bridge-hub-rococo --pallet=pallet_bridge_grandpa
bot bench cumulus-bridge-hubs --runtime=bridge-hub-rococo --pallet=pallet_bridge_parachains
bot bench cumulus-bridge-hubs --runtime=bridge-hub-rococo --pallet=pallet_bridge_messages
bot bench cumulus-bridge-hubs --runtime=bridge-hub-rococo --pallet=pallet_bridge_relayers
bot bench cumulus-bridge-hubs --runtime=bridge-hub-westend --pallet=pallet_bridge_grandpa
bot bench cumulus-bridge-hubs --runtime=bridge-hub-westend --pallet=pallet_bridge_parachains
bot bench cumulus-bridge-hubs --runtime=bridge-hub-westend --pallet=pallet_bridge_messages
bot bench cumulus-bridge-hubs --runtime=bridge-hub-westend --pallet=pallet_bridge_relayers

**Comment by bkontur** [comment]
bot clean
bot bench -v PIPELINE_SCRIPTS_REF=bko-fix cumulus-bridge-hubs --runtime=bridge-hub-rococo --pallet=pallet_bridge_grandpa

**Comment by bkontur** [comment]
bot clean
bot bench -v PIPELINE_SCRIPTS_REF=bko-fix cumulus-bridge-hubs --runtime=bridge-hub-rococo --pallet=pallet_bridge_grandpa

**Comment by bkontur** [comment]
bot clean
bot bench -v PIPELINE_SCRIPTS_REF=bko-fix cumulus-bridge-hubs --runtime=bridge-hub-rococo --pallet=pallet_bridge_grandpa

**Comment by bkontur** [comment]
bot clean
bot bench -v PIPELINE_SCRIPTS_REF=bko-fix cumulus-bridge-hubs --runtime=bridge-hub-rococo --pallet=pallet_bridge_grandpa

**Comment by bkontur** [comment]
bot clean
bot bench -v PIPELINE_SCRIPTS_REF=bko-fix cumulus-bridge-hubs --runtime=bridge-hub-rococo --pallet=pallet_bridge_grandpa

**Comment by bkontur** [comment]
bot clean
bot bench cumulus-bridge-hubs --runtime=bridge-hub-rococo --pallet=pallet_bridge_grandpa

**Comment by bkontur** [comment]
bot bench -v PIPELINE_SCRIPTS_REF=bko-fix cumulus-bridge-hubs --runtime=bridge-hub-rococo --pallet=pallet_bridge_parachains
bot bench -v PIPELINE_SCRIPTS_REF=bko-fix cumulus-bridge-hubs --runtime=bridge-hub-rococo --pallet=pallet_bridge_messages
bot bench -v PIPELINE_SCRIPTS_REF=bko-fix cumulus-bridge-hubs --runtime=bridge-hub-rococo --pallet=pallet_bridge_relayers
bot bench -v PIPELINE_SCRIPTS_REF=bko-fix cumulus-bridge-hubs --runtime=bridge-hub-westend --pallet=pallet_bridge_grandpa
bot bench -v PIPELINE_SCRIPTS_REF=bko-fix cumulus-bridge-hubs --runtime=bridge-hub-westend --pallet=pallet_bridge_parachains
bot bench -v PIPELINE_SCRIPTS_REF=bko-fix cumulus-bridge-hubs --runtime=bridge-hub-westend --pallet=pallet_bridge_messages
bot bench -v PIPELINE_SCRIPTS_REF=bko-fix cumulus-bridge-hubs --runtime=bridge-hub-westend --pallet=pallet_bridge_relayers

File: bridges/primitives/messages/src/source_chain.rs [hunk 3]
```diff
@@ -16,54 +16,47 @@
...
+///
+/// - storage proof of the inbound lane state;
+///
+/// - lane id.
+#[derive(Clone, Decode, Encode, Eq, PartialEq, RuntimeDebug, TypeInfo)]
+pub struct FromBridgedChainMessagesDeliveryProof<BridgedHeaderHash> {
+	/// Hash of the bridge header the proof is for.
+	pub bridged_header_hash: BridgedHeaderHash,
+	/// Storage trie proof generated for [`Self::bridged_header_hash`].
+	pub storage_proof: UnverifiedStorageProof,
```
Comment (bkontur) [inline bridges/primitives/messages/src/source_chain.rs:46]: @serban300 for `FromBridgedChainMessagesProof` we call it 
```
        /// The proved storage containing the messages being delivered.
	pub storage: UnverifiedStorageProof,
```

I would like to align them, what do you prefer here `storage` or `storage_proof`?

Comment (serban300) [reply bridges/primitives/messages/src/source_chain.rs:46]: Hmm I don't have strong preferences. I would say `storage_proof`.

**Comment by bkontur** [comment]
@serban300 can you please help with rebase/merge this branch with `origin/master`?
There are several conflicts with https://github.com/paritytech/polkadot-sdk/pull/4699 :), I mean if you fix just `bridges/primitives/runtime/src/storage_proof.rs`, that would be cool, you don't need to care about weights.

**Comment by bkontur** [comment]
bot clean
bot bench cumulus-bridge-hubs --runtime=bridge-hub-rococo --pallet=pallet_bridge_grandpa
bot bench cumulus-bridge-hubs --runtime=bridge-hub-rococo --pallet=pallet_bridge_parachains
bot bench cumulus-bridge-hubs --runtime=bridge-hub-rococo --pallet=pallet_bridge_messages
bot bench cumulus-bridge-hubs --runtime=bridge-hub-rococo --pallet=pallet_bridge_relayers
bot bench cumulus-bridge-hubs --runtime=bridge-hub-westend --pallet=pallet_bridge_grandpa
bot bench cumulus-bridge-hubs --runtime=bridge-hub-westend --pallet=pallet_bridge_parachains
bot bench cumulus-bridge-hubs --runtime=bridge-hub-westend --pallet=pallet_bridge_messages
bot bench cumulus-bridge-hubs --runtime=bridge-hub-westend --pallet=pallet_bridge_relayers
bot bench cumulus-bridge-hubs --runtime=bridge-hub-rococo --subcommand=xcm --pallet=pallet_xcm_benchmarks::generic
bot bench cumulus-bridge-hubs --runtime=bridge-hub-westend --subcommand=xcm --pallet=pallet_xcm_benchmarks::generic
bot bench cumulus-assets --runtime=asset-hub-rococo --pallet=pallet_xcm_bridge_hub_router
bot bench cumulus-assets --runtime=asset-hub-westend --pallet=pallet_xcm_bridge_hub_router

**Comment by bkontur** [comment]
bot clean
bot bench cumulus-bridge-hubs --runtime=bridge-hub-rococo --pallet=pallet_bridge_grandpa
bot bench cumulus-bridge-hubs --runtime=bridge-hub-rococo --pallet=pallet_bridge_parachains
bot bench cumulus-bridge-hubs --runtime=bridge-hub-rococo --pallet=pallet_bridge_messages
bot bench cumulus-bridge-hubs --runtime=bridge-hub-rococo --pallet=pallet_bridge_relayers
bot bench cumulus-bridge-hubs --runtime=bridge-hub-westend --pallet=pallet_bridge_grandpa
bot bench cumulus-bridge-hubs --runtime=bridge-hub-westend --pallet=pallet_bridge_parachains
bot bench cumulus-bridge-hubs --runtime=bridge-hub-westend --pallet=pallet_bridge_messages
bot bench cumulus-bridge-hubs --runtime=bridge-hub-westend --pallet=pallet_bridge_relayers
bot bench cumulus-bridge-hubs --runtime=bridge-hub-rococo --subcommand=xcm --pallet=pallet_xcm_benchmarks::generic
bot bench cumulus-bridge-hubs --runtime=bridge-hub-westend --subcommand=xcm --pallet=pallet_xcm_benchmarks::generic
bot bench cumulus-assets --runtime=asset-hub-rococo --pallet=pallet_xcm_bridge_hub_router
bot bench cumulus-assets --runtime=asset-hub-westend --pallet=pallet_xcm_bridge_hub_router
